qiita_db/processing_job.py

```diff
@@ -337,3 +507,445 @@ def step(self, value):
                      WHERE processing_job_id = %s"""
             qdb.sql_connection.TRN.add(sql, [value, self.id])
             qdb.sql_connection.TRN.execute()
+
+    @property
+    def children(self):
+        """The children jobs
+
+        Returns
+        -------
+        generator of qiita_db.processing_job.ProcessingJob
+            The children jobs
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT child_id
+                     FROM qiita.parent_processing_job
+                     WHERE parent_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            for jid in qdb.sql_connection.TRN.execute_fetchflatten():
+                yield ProcessingJob(jid)
+
+    def _update_children(self, mapping):
```

Comment:
I broke up this function in 2 pieces - the first one updates the children and returns the ones that are ready for submission (and added a test), the second one submits those jobs. Can't add a tests for this second one (same reasoning as above) but noted in the tests.